### ISSUE TITLE
Port twisted.python.util.FancyStrMixin to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6054]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6054|
|Type|enhancement|
|Created|2012-10-04 16:36:25Z|
|Branch|https://github.com/twisted/twisted/tree/fancystrmixin-py3-6054|

`twisted.python.util.FancyStrMixin` is used by `twisted.names.dns`, and should therefore be ported to Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6054 6054
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_fancystrmixin_py3_6054 branches/fancystrmixin-py3-6054
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349368585000000 1349368585000000
changetime__1349444819000000 1349444819000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/fancystrmixin-py3-6054

I improved docstring, wrote some tests; actual implementation hasn't changed.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [35947]) Branching to 'fancystrmixin-py3-6054'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Thanks.  Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [35959]) Merge fancystrmixin-py3-6054

Author: itamarst
Reviewer: exarkun
Fixes: [#6054](https://github.com/twisted/twisted/issues/6054)

Port `twisted.python.util.FancyStrMixin` to Python 3.
